### PR TITLE
Remove tchannel between agent and collector

### LINKED ISSUE #2105: Remove tchannel reporter from agent
Remove tchannel reporter from agent. 

* agent by default uses `grpc` reporter 
* `tchannel` reporter was deprecated in 1.16 https://github.com/jaegertracing/jaeger/pull/1975.

TODO: 
- [ ] changelog
- [ ] docs
- [ ] remove generated classes

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves  #2105 

Signed-off-by: Pavol Loffay <ploffay@redhat.com>

### CODE REVIEW

File: cmd/agent/main.go [hunk 1]
```diff
@@ -55,11 +54,9 @@ func main() {
 				Namespace(metrics.NSOptions{Name: "agent"})
 
 			rOpts := new(reporter.Options).InitFromViper(v, logger)
-			tchanBuilder := tchannel.NewBuilder().InitFromViper(v, logger)
 			grpcBuilder := grpc.NewConnBuilder().InitFromViper(v)
 			builders := map[reporter.Type]app.CollectorProxyBuilder{
```
Comment (annanay25) [inline cmd/agent/main.go]: Do we need a map here now? We have just one exporter

File: cmd/collector/main.go [hunk 2]
```diff
@@ -36,7 +36,6 @@ import (
 	ss "github.com/jaegertracing/jaeger/plugin/sampling/strategystore"
 	"github.com/jaegertracing/jaeger/plugin/storage"
 	"github.com/jaegertracing/jaeger/ports"
-	tCollector "github.com/jaegertracing/jaeger/tchannel/collector/app"
```
Comment (annanay25) [inline cmd/collector/main.go]: Is this issue also dealing with removing the `tchannel` endpoint from the collector? End user applications will not be able to send spans directly to the collector? This is a breaking change and should be documented

Comment (pavolloffay) [reply cmd/collector/main.go]: End applications are not using this port. This is used only by the agent.

File: cmd/agent/main.go (hunk 1 above)
Comment (pavolloffay) [reply cmd/agent/main.go]: It's needed for extensibility. E.g. at uber they will add custom reporters.

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,24 @@ Changes by Version
 
 ### Backend Changes
```
Comment (yurishkuro) [inline CHANGELOG.md]: Maybe it should go in breaking changes

**Pull request merged**